packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../base/version.dart';
+
+// Include VS Code insiders (useful for debugging).
+const bool _includeInsiders = false;
+
+class VsCode {
+  static const String extensionIdentifier = 'Dart-Code.dart-code';
+
+  VsCode._(this.directory, this.extensionDirectory, { Version version })
+      : this.version = version ?? Version.unknown {
+    _init();
+  }
+
+  final String directory;
+  final String extensionDirectory;
+  final Version version;
+
+  bool _isValid = false;
+  Version extensionVersion;
+  final List<String> _validationMessages = <String>[];
```

Comment:
Since you allow direct access to the list, this could just be public.

(alternatively, if you make the getter return a copy of the internal list, then take care to never modify the contents of the public getter in this class, which you're doing now).

Reply:
In 4f8ebf4 I changed the getter to Iterable and made the internal code add to the internal list. Is this a reasonable fix? There's a slight concern it doesn't prevent the iterable being cast back to a list and mutated?